osu.Game.Rulesets.Mania/ManiaFilterCriteria.cs

```diff
@@ -17,28 +17,58 @@ namespace osu.Game.Rulesets.Mania
 {
     public class ManiaFilterCriteria : IRulesetFilterCriteria
     {
-        private FilterCriteria.OptionalRange<float> keys;
-
+        private FilterCriteria.OptionalRange<float> included_key_range;
+        private HashSet<int> included_keys = new HashSet<int>();
+        private HashSet<int> excluded_keys = new HashSet<int>();
         public bool Matches(BeatmapInfo beatmapInfo, FilterCriteria criteria)
         {
-            return !keys.HasFilter || keys.IsInRange(ManiaBeatmapConverter.GetColumnCount(LegacyBeatmapConversionDifficultyInfo.FromBeatmapInfo(beatmapInfo), criteria.Mods));
+            bool result = (!included_key_range.HasFilter) && (included_keys.Count == 0);
```

Comment:
Please remove parentheses here, this is excessive, the operator precedence here is obvious